### PR TITLE
missing: sort architectures in missing_syscall.h

### PR DESCRIPTION
**Pull request by yuwata**
This also adds several missing syscalls on some architectures.

### CODE REVIEW

File: src/basic/missing_syscall.h [hunk 1]
```diff
@@ -15,22 +15,22 @@
...
 #  define systemd_SC_arch_bias(x) (1024 + (x))
-#elif defined __alpha__
-#  define systemd_SC_arch_bias(x) (110 + (x))
-#elif defined _MIPS_SIM
-#  if _MIPS_SIM == _MIPS_SIM_ABI32
+#elif defined(_MIPS_SIM)
+#  if (_MIPS_SIM) == (_MIPS_SIM_ABI32)
 #    define systemd_SC_arch_bias(x) (4000 + (x))
-#  elif _MIPS_SIM == _MIPS_SIM_NABI32
+#  elif (_MIPS_SIM) == (_MIPS_SIM_NABI32)
```
Comment (keszybz) [inline src/basic/missing_syscall.h]: Hmm, why the new parens?

**Review by keszybz — COMMENTED** [review]
Would it be possible to split the sorting and the additions? I would like to verify the additions, but finding them is hard...

File: src/basic/missing_syscall.h (hunk 1 above)
Comment (yuwata) [reply src/basic/missing_syscall.h]: dropped.

**Comment by yuwata** [comment]
@keszybz I've split the commit into 5. PTAL.

File: src/basic/missing_syscall.h [hunk 2]
```diff
@@ -134,7 +134,7 @@ static inline int missing_memfd_create(const char *name, unsigned int flags) {
 #  endif
 #elif defined(__powerpc__)
 #  define systemd_NR_getrandom 359
-#elif defined(__s390x__)
+#elif defined(__s390__)
```
Comment (poettering) [inline src/basic/missing_syscall.h:143]: you are saying `__s390__` is always set if `__s390x__`, did I get this right?

File: src/basic/missing_syscall.h [hunk 3]
```diff
@@ -53,32 +53,38 @@ static inline int missing_pivot_root(const char *new_root, const char *put_old)
...
 #  define systemd_NR_memfd_create 279
-#elif defined __alpha__
+#elif defined(__alpha__)
 #  define systemd_NR_memfd_create 512
-#elif defined(__powerpc__)
-#  define systemd_NR_memfd_create 360
-#elif defined __s390__
-#  define systemd_NR_memfd_create 350
-#elif defined _MIPS_SIM
+#elif defined(__arc__) || defined(__tilegx__)
```
Comment (poettering) [inline src/basic/missing_syscall.h:60]: are these archs actually related? or is this coincidence?

File: src/basic/missing_syscall.h [hunk 4]
```diff
@@ -334,7 +334,7 @@ static inline pid_t raw_getpid(void) {
 #  elif _MIPS_SIM == _MIPS_SIM_ABI64
 #    define systemd_NR_renameat2 systemd_SC_arch_bias(311)
 #  endif
-#elif defined(__powerpc64__)
+#elif defined(__powerpc__)
```
Comment (poettering) [inline src/basic/missing_syscall.h:361]: so, `__powerpc__` is always set when `__powerpc64__` is set, too, did I get this right?

**Review by poettering — APPROVED** [review]
looks good to me, but didn't check the syscall nrs

File: src/basic/missing_syscall.h (hunk 2 above)
Comment (yuwata) [reply src/basic/missing_syscall.h:143]: Right.

File: src/basic/missing_syscall.h (hunk 3 above)
Comment (yuwata) [reply src/basic/missing_syscall.h:60]: All syscall numbers for these architectures are equivalent, and they have does not have specific numbers. The syscall numbers defined in include/uapi/asm-generic/unistd.h are used for those archs.

File: src/basic/missing_syscall.h (hunk 4 above)
Comment (yuwata) [reply src/basic/missing_syscall.h:361]: Right.
